### DISCUSSION TITLE
Possibility to change the Java version

### DISCUSSION THREAD

**Discussion by filipbart** [codemagic-ci-cd/codemagic-docs#2132] [Q&A] [answered]
### Use case

```Text
I'm using the gradle version that requires Java 17, the default version on machine is 11. There is no option to choose the java version for build on Workflow editor. The only solution is to add a codemagic.yaml file and specify the java version there.
```

### Idea

```Text
On Workflow editor in build section, add the same select option like "Flutter or Xcode version" but with Java version.
```

**Comment by mtcn** [comment]
Hi, you should be able to change java version like this inside codemagic.yaml file

<img width="330" alt="Screenshot 2023-10-23 at 19 06 26" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/13784221/525a06c2-5e34-4231-8664-6634bb4c5218">

**Reply by filipbart** [reply]
Hi, yes. But wouldn't I then have to give up the workflow editor? And specify everything inside codemagic.yaml?

**Reply by mtcn** [reply, marked as answer]
sorry, I misunderstood your use case. I haven't used Codemagic with the workflow editor but have you tried to add the Java version to the environment variables section?

JAVA_HOME = /usr/lib/jvm/java-1.17.0-openjdk-amd64  (path is mentioned in [here](https://docs.codemagic.io/specs/versions-linux/#java-versions) )
or 
java = 17

**Reply by maxstubbersfield** [reply]
Thanks @mtcn! 

Also just to add, when setting JAVA_HOME the path will depend on the OS of the machine you are using :)

Here are the relevant links:
[Windows](https://docs.codemagic.io/specs/versions-windows/#java-versions)
[MacOS](https://docs.codemagic.io/specs/versions-macos/#java-versions)
[Linux](https://docs.codemagic.io/specs/versions-linux/#java-versions)

**Comment by himesh-cm** [comment]
Thank you, @mtcn, for pitching in.
@filipbart - While I’m closing our discussion for now, if you have any other questions or suggestions, feel free to reply back or start a new discussion anytime. Have a nice day!

**Comment by samnbd** [comment]
@himesh-cm what about Java 18...I'm facing an issue when trying to build my Flutter app where I use [nordic_dfu](https://pub.dev/packages/nordic_dfu) and when I try to build the app for Android...Code magic will throw the error below: 

```
Execution failed for task ':nordic_dfu:compileReleaseJavaWithJavac'.
> error: invalid source release: 17
```

When I build the app locally on my machine, I don't have this issue but I'm using Java 18

**Reply by behzodfaiziev** [reply]
facing the same issue using latest packages of ```Flutter Community Plus```

**Reply by david-legend** [reply]
Inside code magic, create a new env variable called JAVA_HOME and assign this value /Library/Java/JavaVirtualMachines/zulu-17.jdk/Contents/Home


note you can choose any version of java from this list

https://docs.codemagic.io/specs/versions-macos/

In this particular example it requires Java 17

**Comment by sampaio96** [comment]
I needed to add this script after saving the environment variables:

```
flutter config --jdk-dir $JAVA_HOME
```

Before doing this, `flutter doctor -v` was not showing the right Java version on the Android toolchain.

(Originally in #3035)
